### ISSUE TITLE
Chrome/Opera/Chromium-only bug (not Safari WebKit): cover image "fit" overflows in next column

### ISSUE THREAD

**Issue by danielweck** [readium/readium-shared-js#144]
EPUB:
https://readiumfoundation.box.com/s/eruhof2djmz3mmj1qg3a

**Comment by danielweck** [comment]
Related: https://github.com/readium/readium-shared-js/issues/138 and https://github.com/readium/readium-shared-js/issues/88

**Comment by JayPanoz** [comment]
> The reason is that the html element's computed CSS style reports a scrollWidth that is larger than the actual width of its body.

We had this bug in an early version of Readium CSS since we had to find a way to support images with `position: absolute; bottom: 0` on a title page in one of our samples—it turns out some authors don’t bother setting an explicit `height` and `position: relative` for the parent and think `position: absolute` for the `img` will automagically work.

TL;DR: because of margin collapse, body’s margin was the first element’s (`h1`) so although the `height` was `100%`, there was this margin creating another column in single-page view. 

As mentioned, setting an explicit `height` to the `body` element wouldn’t solve all issues though, since `%` values depends on the parent element so if there is an extra `section`, `figure` or `div` in the markup, the image would still overflow if the author is using `height: 100%` for instance. The rendering engine would not be able to compute a value for `100%` since some of its parent elements would not have an explicit `height`. 

Consequently, we ended up creating [safeguards for media elements](https://github.com/readium/readium-css/blob/phase3-user_settings/readiumCSS/1-pagination/readiumCSS-safeguards.css#L54), with a `max-height` set in `vh`. It’s like `%` but it depends on the viewport height, not the parent element’s. [Support is excellent](http://caniuse.com/#feat=viewport-units).

Please note it’s `95vh` to account for an hypothetical `figcaption` or `margin-top`, some authors nesting `img` in paragraphs….

You might want to get rid of this safeguard in scroll view though. I know I had some weird issues in EPUB files when using `vh` for images.

We’re also using `object-fit: contain;` to keep the correct aspect-ratio but this one [won’t work in IE11 and Edge < 16](http://caniuse.com/#feat=object-fit).

There are polyfills available though: 

- [Fitie](https://github.com/jonathantneal/fitie)
- [Object fit images](https://github.com/bfred-it/object-fit-images)
- [Object fit polyfill](https://github.com/constancecchen/object-fit-polyfill) (this one supports `picture` and `video` as well)
- Another [object fit polyfill](https://github.com/tonipinel/object-fit-polyfill)

**Comment by danielweck** [comment]
Useful update, thanks @JayPanoz
